ebi_eva_common_pyutils/variation/contig_utils.py

```diff
@@ -0,0 +1,92 @@
+    response_lines = response.content.decode("utf-8").split("\n")
+    num_lines = len(response_lines)
+
+    features_section_found, source_line_found = False, False
+    chosen_response = []
+    line_index = 0
+    # Look for the "source" feature under the "Features" section in the text response
+    while line_index < num_lines:
+        line = response_lines[line_index]
+        if not (features_section_found or line.lower().startswith("fh   key")):
+            line_index += 1
+            continue
+        features_section_found = True
+        # Based on "Data item positions" described here, http://www.insdc.org/files/feature_table.html#3.4.2
+        # the sixth character represents the start of the feature key
+        if not (source_line_found or line[5:].lower().startswith("source")):
+            line_index += 1
+            continue
+        source_line_found = True
+        if line[21:].startswith("/"):
+            assembled_line = line.strip()
+            line_index += 1
+            # Assemble text spread across multiple lines until
+            # we hit the next qualifier (starts with /) or the next section
+            while line_index < num_lines and \
+                    not (response_lines[line_index][21:].startswith("/")
+                         or response_lines[line_index][5:6].strip() != ''):
+                line = response_lines[line_index]
+                assembled_line += " " + line[21:].strip()
+                line_index += 1
+
+            # Fall back to organelle in case of MT/Chloroplast accessions
+            # and the reference notes in case of Linkage Group molecules
+            chosen_response = re.findall('.*/chromosome=".+"', assembled_line) or \
+                              re.findall('.*/organelle=".+"', assembled_line) or \
+                              re.findall('.*/note=".+"', assembled_line)
+
+            # If we have a response to give, no need to continue further
+            # If the sixth character is not empty, we have reached the next feature, so no need to continue further
+            if chosen_response or line[5:6].strip() != '':
+                break
+        else:
+            line_index += 1
+
+    if not chosen_response:
+        return ""
+
+    return str.split(chosen_response[0], '"')[1].strip()
+
+
+def is_wgs_accession_format(contig_accession):
+    wgs_prefix = contig_accession[:4]
+    wgs_numeric_suffix = contig_accession[4:].replace(".", "")
+    return str.isalpha(wgs_prefix) and str.isnumeric(wgs_numeric_suffix)
+
+
+def get_chromosome_name_for_contig_accession(contig_accession):
+    # Don't bother calling the ENA web service to get the chromosome number if the accession is a WGS accession
+    # since the API will proceed to download the entire WGS dataset which can be in hundreds of MBs or even GBs
+    # See https://www.ebi.ac.uk/ena/browser/api/text/AABR07050911.1?lineLimit=100&annotationOnly=true for example
```

Comment:
```suggestion
    """
    Don't bother calling the ENA web service to get the chromosome number if the accession is a WGS accession
    since the API will proceed to download the entire WGS dataset which can be in hundreds of MBs or even GBs
    See https://www.ebi.ac.uk/ena/browser/api/text/AABR07050911.1?lineLimit=100&annotationOnly=true for example
    """
```

Reply:
This is just a comment and doesn't document the function which essentially gets a chromosome name corresponding to a Genbank Contig.